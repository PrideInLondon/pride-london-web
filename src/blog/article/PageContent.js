import React from 'react'
import PropTypes from 'prop-types'
import BlogHeader from './BlogHeader'
import { BlogContent } from './blogContent'
import BlogFooter from './BlogFooter'

const PageContent = ({ article, title, datePublished, category, author }) => (
  <>
    <BlogHeader
      title={title}
      datePublished={datePublished}
      category={category}
    />
    <BlogContent article={article} />
    <BlogFooter author={author} />
  </>
)

PageContent.propTypes = {
  article: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
  datePublished: PropTypes.string.isRequired,
  category: PropTypes.object.isRequired,
  author: PropTypes.object,
}
PageContent.defaultProps = {
  author: null,
}

export default PageContent
