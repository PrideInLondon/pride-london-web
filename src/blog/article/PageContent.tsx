import React from 'react'
import { Wrapper } from '../../components/wrapper/Wrapper'
import { BlogHeader } from './BlogHeader'
import { BlogWrapper } from './BlogHeader.styles'
import { BlogContent } from './blogContent'
import { BlogFooter } from './BlogFooter'
import { ShareBarWrapper, DesktopShareBar } from './PageContent.styles'
import { PageContentProps } from './PageContent.types'

export const PageContent: React.FC<PageContentProps> = ({
  article,
  title,
  datePublished,
  category,
  author,
}) => {
  const availableContent =
    article.json.content[0].content[0] &&
    article.json.content[0].content[0].nodeType === 'text'
      ? article.json.content[0].content[0].value
      : ''
  const body = JSON.stringify(availableContent)
  let location = ''
  if (typeof window !== 'undefined') {
    location = window.location.href
  }
  return (
    <>
      <BlogHeader
        title={title}
        datePublished={datePublished}
        category={category}
        body={body}
      />
      <Wrapper position="relative">
        <ShareBarWrapper>
          <DesktopShareBar
            variant="vertical"
            content={{
              title,
              body: body,
              url: location,
            }}
          />
        </ShareBarWrapper>
        <BlogWrapper>
          <BlogContent article={article} />
          <BlogFooter author={author} />
        </BlogWrapper>
      </Wrapper>
    </>
  )
}
