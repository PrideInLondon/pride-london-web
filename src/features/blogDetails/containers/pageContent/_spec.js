import React from 'react'
import { shallow } from 'enzyme'
import PageContent from '.'

describe(PageContent.name, () => {
  it('renders with default values', () => {
    //article, title, datePublished, category, author
    const title = 'Title'
    const article = {}
    const datePublished = '2019-03-28T20:30+00:00'
    const category = ['Resaearch']
    const wrapper = shallow(
      <PageContent
        title={title}
        article={article}
        category={category}
        datePublished={datePublished}
        author={null}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
