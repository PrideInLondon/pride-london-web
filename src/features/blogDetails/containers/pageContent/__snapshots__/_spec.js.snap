// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContent renders with default values 1`] = `
<Fragment>
  <BlogHeader
    category={
      Object {
        "hexColour": "#eeeeee",
        "title": "Resaearch",
      }
    }
    datePublished="2019-03-28T20:30+00:00"
    readLength={null}
    title="Title"
  />
  <BlogContent
    article={Object {}}
  />
  <BlogFooter
    author={null}
  />
</Fragment>
`;
